internal/provider/clickhouse_cluster_resource.go

```diff
@@ -664,10 +697,22 @@ func (m *clickhouseClusterResources) parse(rs *clickhouse.ClusterResources) diag
 			m.Keeper = nil
 		} else {
 			m.Keeper = new(clickhouseClusterResourcesKeeper)
-			m.Keeper.ResourcePresetId = types.StringValue(v.ResourcePresetId)
+			m.Clickhouse.ResourcePresetId = types.StringValue(v.ResourcePresetId)
```

Comment:
You still set `ClickHouse.ResourcePresetId = Keeper.ResourcePresetId`.
It's a typo.